### PR TITLE
JSON-API serialization

### PR DESCRIPTION
**Pull request by nono**
It should be enough for the files API (serialization only, unmarshalling in a future PR).

### CODE REVIEW

File: web/jsonapi/data.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	"github.com/cozy/cozy-stack/couchdb"
+)
+
+// Object is an interface to serialize something to a JSON-API Object
+type Object interface {
+	couchdb.Doc
+	SelfLink() string
+	Relationships() RelationshipMap
+	Included() []Object
+}
```
Comment (jinroh) [inline web/jsonapi/data.go]: Wouldn't it be useful to inherit the `json.Marshaler` interface here ?

**Comment by jinroh** [comment]
LGTM 👍

File: web/jsonapi/data.go (hunk 1 above)
Comment (nono) [inline web/jsonapi/data.go]: It's not in couchdb.Doc?

Comment (jinroh) [inline web/jsonapi/data.go]: Nope. I think it makes sense to put it in `jsonapi.Object` to make it clear that we are talking about the serialisation for the public API side (and not couch for instance)...

File: web/jsonapi/jsonapi.go [hunk 2]
```diff
@@ -12,18 +12,43 @@ import (
...
+	if err != nil {
+		AbortWithError(c, InternalServerError(err))
+		return
+	}
+	doc := Document{
+		Data:     &data,
+		Links:    links,
+		Included: included,
+	}
+	body, err := json.Marshal(doc)
```
Comment (jinroh) [inline web/jsonapi/jsonapi.go]: Also, the `included` elements are not serialized adequatly here.

File: web/jsonapi/data.go (hunk 1 above)
Comment (nono) [inline web/jsonapi/data.go]: Well, in fact, it looks like `struct`s don't implement the `json.Marshaler` interface, even if we can serialize them to JSON.

**Pull request merged**